testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -1280,6 +1280,24 @@ public void testDeletePartitionedTable()
         });
     }
 
+    @Test(groups = HIVE_TRANSACTIONAL, timeOut = TEST_TIMEOUT)
+    public void testUpdatePartitionedTable()
+    {
+        withTemporaryTable("update_partitioned", true, true, NONE, tableName -> {
+            onHive().executeQuery(format("CREATE TABLE %s (nationkey bigint, name string) PARTITIONED BY (regionkey bigint) " +
+                    "STORED AS ORC TBLPROPERTIES ('transactional' = 'true')", tableName));
+            onPresto().executeQuery(format("INSERT INTO %s SELECT nationkey, name, regionkey FROM tpch.tiny.nation", tableName));
+
+            verifySelectForPrestoAndHive("SELECT COUNT(*) FROM " + tableName, "nationkey < 100", row(25));
+
+            onPresto().executeQuery(format("UPDATE %s SET nationkey = nationkey + 333 WHERE regionkey = 4 AND nationkey %% 10 = 3", tableName));
```

Comment:
You can use `nationkey < 100` to avoid `%%`.
it'd expose the bug the same way.

Reply:
That would match all rows, rather than specific ones. You're right that it would expose the bug, I feel like updating all rows is somewhat of a special case and thus a different test.

Reply:
Removed the test